File: src/main/java/org/antop/java/chapter5/GroupBy.java
package org.antop.java.chapter5;

import org.antop.kotlin.chapter5.Person;

import java.util.List;
import java.util.stream.Collectors;

public class GroupBy {
    public static void main(String[] args) {
        var people = List.of(
                new Person("Alice", 31),
                new Person("Bob", 29),
                new Person("Carol", 31)
        );
        var map = people.stream()
                .collect(Collectors.groupingBy(Person::getAge));
        System.out.println(map);

        var counter = people.stream()
                .collect(Collectors.groupingBy(Person::getAge, Collectors.counting()));
        System.out.println(counter);
    }
}

File: src/main/kotlin/org/antop/kotlin/chapter5/groupBy.kt
package org.antop.kotlin.chapter5

fun main() {
    val people = listOf(
        Person("Alice", 31),
        Person("Bob", 29),
        Person("Carol", 31),
    )
    val map = people.groupBy { it.age }
    println(map) // {31=[Person(name=Alice, age=31), Person(name=Carol, age=31)], 29=[Person(name=Bob, age=29)]}

    val counter = people.groupingBy { it.age }.eachCount()
    println(counter)
}
